### ISSUE TITLE
Connection pool configuration

### ISSUE THREAD

**Issue by pstarace** [ckan/ckan#2232]
Hi all

I found that the connection pool for sqlalchemy is instantiated in ckan/config/environment.py, row 355:
```
engine = sqlalchemy.engine_from_config(config, 'sqlalchemy.', **extras)
```
So it should be feasible to inject pooling configuration adding new properties in the ini config file, something like:
```
sqlalchemy.pool_size=10
sqlalchemy.max_overflow=20
```
Can you please confirm?

I'd like also to do the same for datastore, but I found the current code in ckan/ckanext/datastore/db.py doesn't support extra configuration:
```
engine = sqlalchemy.create_engine(connection_url)
```
In this case the pool is created using the default params (pool_size=5, max_overflow=10).
Do you know any drawback adding new properties in the ini file:
```
ckan.datastore.pool_size=10
ckan.datastore.max_overflow=20
```
and loading them using:
```
import pylons
try:
    extras = {'pool_size': int(pylons.config['ckan.datastore.pool_size']), 
        'max_overflow' : int(pylons.config['ckan.datastore.max_overflow'])}
except Exception:
    extras = {'pool_size': 5, 'max_overflow' : 10}
engine = sqlalchemy.create_engine(connection_url, **extras)
```
?

Thanks
Paolo

**Comment by davidread** [comment]
Yes I believe you can pass in parameters to the sqla enginer using config settings like `sqlalchemy.pool_size`. I remember playing with the pool like this before successfully. (The one that didn't work was `poolclass` as you can only provide a string in the config file, rather than an actual class, but it sounds like you're not interested in that).

It sounds like a good plan to do the same for datastore. However I'd suggest adding .sqlalchemy in the keys, to keep these options in their own namespace. So you'd have: `ckan.datastore.sqlalchemy.pool_size`
And rather than catch all exceptions it would be better to catch a specific one. Or you could perhaps just use pylons.config.get and supply the default. Anyway, please do submit a Pull Request.

**Comment by amercader** [comment]
Fixed in #2244
